tests/unit_test/events/definitions_test.py

```
@@ -1199,3 +1198,136 @@ def test_delete_message_broadcast(self):
+    @responses.activate
+    def test_validate_mail_channel_schedule_event(self):
+        from kairon.events.definitions.mail_channel import MailProcessEvent
+        bot = "test_add_schedule_event"
+        user = "test_user"
+        url = f"http://localhost:5001/api/events/execute/{EventClass.mail_channel_process_mails}?is_scheduled=False"
+        responses.add(
+            "POST", url,
+            json={"message": "test msg", "success": True, "error_code": 400, "data": None}
+        )
+        with patch('kairon.shared.channels.mail.processor.MailProcessor.__init__', return_value=None) as mp:
+            with patch('kairon.shared.channels.mail.processor.MailProcessor.login_smtp', return_value=None) as mock_login:
```

Ruff (0.8.0)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)